### PR TITLE
User Login

### PR DESCRIPTION
**Pull request by TELBC**
### What was done?
- fixed the .gitignore to actually ignore .idea folders
- added Authcontroller for Registration and Login
- added UserController for profile updating, deleting and password changing & preference updating
- changed Model to have a Password field(Hashed when saved)
- added JWT Token authentication
- All endpoints except for Registration and Login are protected (need bearer token to access)
- added Tests
    - Tests for Service classes
    - Tests for Controllers/Endpoints (Test coverage is not very extensive, am having problems with the Authorization, need to ask)

#### TLDR:
Added these endpoints with JWT authentication:
image

### CODE REVIEW

**Comment by XenoxJeager** [comment]
looks good but DeleteUserModel is a dto right so add it to the DTO folder not in UserController?

**Review by DT-school — APPROVED** [review]
